Editor/Utility/CinemachineSceneTools.cs

```diff
@@ -0,0 +1,564 @@
+
+                if (!cmToolbarIsHidden)
+                {
+                    // only display cm tools that are relevant for the current selection
+                    foreach (var toolHandle in s_ExclusiveTools)
+                    {
+                        toolHandle.Value.IsDisplayedSetter(s_RequiredTools.ContainsKey(toolHandle.Key));
+                    }
+                    foreach (var toolHandle in s_Tools)
+                    {
+                        toolHandle.Value.IsDisplayedSetter(s_RequiredTools.ContainsKey(toolHandle.Key));
+                    }
+                }
+                
+                if (s_Tools.ContainsKey(s_SoloVcamToolType)) {
+                    s_Tools[s_SoloVcamToolType].ToggleSetter(CinemachineBrain.SoloCamera != null);
+                }
+            };
+        }
+        static Type s_SoloVcamToolType = typeof(SoloVcamTool);
+    }
+    
+    static class CinemachineSceneToolHelpers
+    {
+        const float k_DottedLineSpacing = 4f;
+        public const float lineThickness = 4f;
+
+        static GUIStyle s_LabelStyle = new GUIStyle 
+        { 
+            normal =
+            {
+                background = AssetDatabase.LoadAssetAtPath<Texture2D>(ScriptableObjectUtility.
+                    CinemachineRealativeInstallPath + "/Editor/EditorResources/SceneToolsLabelBackground.png"),
+                textColor = Handles.selectedColor,
+            },
+            fontStyle = FontStyle.Bold,
+            padding = new RectOffset(5, 0, 5, 0)
+        };
+        
+
+        public static float SliderHandleDelta(Vector3 newPos, Vector3 oldPos, Vector3 forward)
+        {
+            var delta = newPos - oldPos;
+            var sameDirection = Vector3.Dot(delta.normalized, forward) > 0;
+            return (sameDirection ? 1f : -1f) * delta.magnitude;
+        }
+
+        /// <summary>
+        /// Calculate delta and discard imprecision.
+        /// </summary>
+        public static Vector3 PositionHandleDelta(Quaternion rot, Vector3 newPos, Vector3 oldPos)
+        {
+            var delta =
+                Quaternion.Inverse(rot) * (newPos - oldPos);
+            delta = new Vector3(
+                Mathf.Abs(delta.x) < UnityVectorExtensions.Epsilon ? 0 : delta.x,
+                Mathf.Abs(delta.y) < UnityVectorExtensions.Epsilon ? 0 : delta.y,
+                Mathf.Abs(delta.z) < UnityVectorExtensions.Epsilon ? 0 : delta.z);
+            return delta;
+        }
+        
+        public static void DrawLabel(Vector3 position, string text)
+        {
+            var labelOffset = HandleUtility.GetHandleSize(position) / 5f;
+            Handles.Label(position + new Vector3(0, -labelOffset, 0), text, s_LabelStyle);
+        }
+
+        static int s_ScaleSliderHash = "ScaleSliderHash".GetHashCode();
+        public static void FovToolHandle(CinemachineVirtualCameraBase vcam, ref LensSettings lens, 
+            bool isLensHorizontal, ref float fov, ref bool soloSetByTools)
```

Comment:
Why are you passing in lens _and_ fov?  Isn't lens enough?  Callers have a bunch of logic to make fov from lens.  Can't that be moved here and the calling code simplified?

Reply:
I changed the name to _fovAfterLastToolModification_, because it was not clear at all what this fov was doing. This fov value is needed to reverse the ScaleHandles effect. This fov represents, as the name suggests now, the value of fov after the tool moved the tool.

Reply:
I don't see why you need the callers to manage it.  Can't you manage it locally with a static variable?